wallet/src/signer/ledger_signer/tests.rs

```diff
@@ -0,0 +1,414 @@
+                handle.button(Button::Both, Action::PressAndRelease).await.unwrap();
+            }
+            msg = control_msg_rx.recv() => {
+                match msg {
+                    Some(ControlMessage::Finish) => {
+                        eprintln!("Received finish signal.");
+                        break;
+                    }
+                    None => {
+                        eprintln!("Channel closed.");
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    println!("Auto-confirmer task finished.");
+}
+
+struct DummyProvider;
+
+#[async_trait]
+impl LedgerFinder for DummyProvider {
+    type Ledger = TcpDevice;
+
+    async fn find_ledger_device_from_db<T: WalletStorageReadLocked + Send>(
+        &self,
+        _db_tx: &mut T,
+        _chain_config: Arc<ChainConfig>,
+    ) -> SignerResult<(Self::Ledger, LedgerData)> {
+        Err(SignerError::LedgerError(LedgerError::NoDeviceFound))
+    }
+}
+
+async fn setup(
+    deterministic_aux: bool,
+) -> (
+    tokio::task::JoinHandle<()>,
+    Sender<ControlMessage>,
+    impl Fn(Arc<ChainConfig>, U31) -> LedgerSigner<TcpDevice, DummyProvider>,
+) {
+    let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 5001);
+    let handle = PodmanHandle::new(addr);
+
+    let mut transport = TcpTransport::new().unwrap();
+    let mut device = transport
+        .connect(TcpInfo {
+            addr: SocketAddr::new(std::net::IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 9999),
+        })
+        .await
+        .unwrap();
+
+    let mut tries = 0;
+    loop {
+        match get_app_name(&mut device).await {
+            Ok(_) => break,
+            Err(_) => {
+                tries += 1;
+                if tries > 10 {
+                    break;
+                }
+                sleep(Duration::from_millis(100)).await;
+            }
+        }
+    }
+
+    let device = Arc::new(Mutex::new(device));
+
+    let (control_msg_tx, control_msg_rx) = mpsc::channel(1);
+    let auto_clicker = tokio::spawn(auto_confirmer(control_msg_rx, handle));
+
+    (auto_clicker, control_msg_tx, move |chain_config, _| {
+        let aux_provider: Box<dyn SigAuxDataProvider + Send> = if deterministic_aux {
+            Box::new(PredefinedSigAuxDataProvider)
+        } else {
+            Box::new(make_true_rng())
+        };
+
+        LedgerSigner::new_with_sig_aux_data_provider(
+            chain_config,
+            device.clone(),
+            aux_provider,
+            DummyProvider {},
+        )
+    })
+}
+
+#[rstest]
+#[trace]
+#[serial_test::serial]
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_app_name() {
+    let mut transport = TcpTransport::new().unwrap();
+    let mut device = transport
+        .connect(TcpInfo {
+            addr: SocketAddr::new(std::net::IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 9999),
```

Comment:
Plz don't duplicate the port and the address, factor out the connection creation instead.